platform-sdk/swirlds-merkledb/src/jmh/java/com/swirlds/benchmark/DataFileWriterBenchmark.java

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: Apache-2.0
+package com.swirlds.benchmark;
+
+import static com.swirlds.common.test.fixtures.DataUtils.randomUtf8Bytes;
+
+import com.hedera.pbj.runtime.io.buffer.BufferedData;
+import com.swirlds.common.io.utility.FileUtils;
+import com.swirlds.merkledb.files.DataFileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import org.openjdk.jmh.annotations.*;
+
+/**
+ * This benchmarks can be used to measure performance of {@link DataFileWriter} class. <p>
+ * The following parameters are used to configure the benchmark:
+ * <ul>
+ *   <li><b>bufferSizeMb</b>: Size of the buffer in MB.</li>
+ *   <li><b>maxFileSizeMb</b>: Maximum size of the file to write in MB.</li>
+ *   <li><b>sampleSize</b>: Number of sample data items to generate.</li>
+ *   <li><b>sampleRangeBytes</b>: Range of the sample data items in bytes, chosen randomly from the sample.</li>
+ * </ul>
+ */
+@State(Scope.Benchmark)
+public class DataFileWriterBenchmark {
+
+    private static final int MEGABYTE = 1024 * 1024;
+
+    @Param({"16", "64", "128", "256"})
+    public int bufferSizeMb;
+
+    @Param({"50", "200", "500"})
+    public int maxFileSizeMb;
+
+    @Param({"10"})
+    public int sampleSize;
+
+    // first is test for small data items like hashes
+    @Param({"56-56", "300-1000"})
+    public String sampleRangeBytes;
+
+    // Runtime variables
+    private Random random;
+    private BufferedData[] sampleData;
+    private long maxFileSize;
+
+    private Path benchmarkDir;
+    private DataFileWriter dataFileWriter;
+
+    @Setup(Level.Trial)
+    public void setupGlobal() throws IOException {
+        // Initialize random generator
+        random = new Random(1234);
+        maxFileSize = maxFileSizeMb * MEGABYTE;
+        benchmarkDir = Files.createTempDirectory("dataFileWriterBenchmark");
+
+        // Generate sample data
+        String[] range = sampleRangeBytes.split("-");
+        int sampleMinLength = Integer.parseInt(range[0]);
+        int sampleMaxLength = Integer.parseInt(range[1]);
+
+        sampleData = new BufferedData[sampleSize];
+        for (int i = 0; i < sampleSize; i++) {
+            sampleData[i] = BufferedData.wrap(randomUtf8Bytes(random.nextInt(sampleMinLength, sampleMaxLength + 1)));
```

Comment:
Is there a way to enhance the benchmark to use native buffers, not only Java heap buffers?